### PR TITLE
[fix](csv-parsing) Reset state every time after extending output buffer

### PR DESCRIPTION
**Pull request by Hastyshell**
### What problem does this PR solve?

Multi char line delimiter or column seperator may be truncate when large csv file ingestion, this is a known problem.

Previously, for performance reason, we take the risk. However, we should take it into consideration because this case became a real problem.

I also propose to refactor the csv reader and introduce a more elegant state machine for csv parsing in future version.

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

File: be/src/vec/exec/format/file_reader/new_plain_text_line_reader.cpp [hunk 1]
```diff
@@ -325,13 +324,13 @@ Status NewPlainTextLineReader::read_line(const uint8_t** ptr, size_t* size, bool
...
         return Status::OK();
     }
-    _line_reader_ctx->refresh();
     int found_line_delimiter = 0;
     size_t offset = 0;
     bool stream_end = true;
     while (!done()) {
         // find line delimiter in current decompressed data
         uint8_t* cur_ptr = _output_buf + _output_buf_pos;
+        _line_reader_ctx->refresh();
```
Comment (TangSiyang2001) [inline be/src/vec/exec/format/file_reader/new_plain_text_line_reader.cpp:333]: This fix may result in bad case in very wide table, aka per line is larger than read buffer size.
We previously not did so for the parse error case is low probability and for performance reason.

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Review by gavinchou — APPROVED** [review]

**Pull request closed without merging**